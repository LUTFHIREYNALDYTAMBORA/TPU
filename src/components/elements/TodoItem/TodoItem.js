import React from 'react';
import PropTypes from 'prop-types';

export default function TodoItem(props) {
  const { todo, onRemove } = props;

  return (
    <li className="flex bg-gray-100 p-3 my-2">
      <button
        className="mr-2 font-bold text-red-400"
        onClick={() => onRemove(todo)}
      >
        X
      </button>
      <p>{todo.text}</p>
    </li>
  );
}

TodoItem.defaultProps = {
  onRemove: () => {},
  todo: {}
};

TodoItem.propTypes = {
  onRemove: PropTypes.func,
  todo: PropTypes.object
};
