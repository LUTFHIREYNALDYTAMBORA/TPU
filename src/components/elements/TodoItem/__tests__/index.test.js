import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TodoItem from '../TodoItem';

describe('src/components/elements/TodoItem', () => {
  test('render correctly', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<TodoItem />);
    expect(tree).toMatchSnapshot();
  });
});
