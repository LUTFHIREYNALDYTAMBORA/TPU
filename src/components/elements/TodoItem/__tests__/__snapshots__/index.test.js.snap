// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/TodoItem render correctly 1`] = `
<li
  className="flex bg-gray-100 p-3 my-2"
>
  <button
    className="mr-2 font-bold text-red-400"
    onClick={[Function]}
  >
    X
  </button>
  <p />
</li>
`;
